fdb-record-layer-core/src/main/proto/record_metadata.proto

```diff
@@ -97,99 +99,12 @@ message MetaData {
   optional bool uses_subspace_key_counter = 11;
   repeated JoinedRecordType joined_record_types = 12;
   repeated UnnestedRecordType unnested_record_types = 13;
+  repeated Udf udfs = 14;
```

Comment:
I'm trying to think a little bit about how we'd evolve this forward in the future. The feature implemented here is for scalar valued functions, which works pretty well with our `Value` abstraction. If we wanted to implement table-valued functions I think we'd need slightly different machinery, and I'm trying to think if we'd want to have one "UDF" message that had a oneof over the different function types, or if we'd want to have different `ScalarValuedFunction` and `TableValuedFunction` messages at this level.

My inclination is to think that you'd want them to be separate, which makes me think that we'd want to call the `Udf` message here something like `ScalarValuedFunction` in anticipation of some future function types

Reply:
I agree, I changed the `Udf` to be `ScalarValuedFunction`.